templates/alertmanagerconfig.go

```diff
@@ -31,17 +31,43 @@ func convertToApiExtV1JSON(val interface{}) apiextensionsv1.JSON {
 	return out
 }
 
+var _true = true
+
+var smtpRegxRuleMatchers = "CephClusterNearFull" +
+	"|CephClusterCriticallyFull" +
+	"|CephClusterReadOnly" +
+	"|CephOSDCriticallyFull" +
+	"|CephOSDNearFull" +
+	"|PersistentVolumeUsageNearFull" +
+	"|PersistentVolumeUsageCritical"
+
+var pagerdutyRegxRuleMatchers = "critical|warning"
+
 var AlertmanagerConfigTemplate = promv1a1.AlertmanagerConfig{
 	Spec: promv1a1.AlertmanagerConfigSpec{
 		Route: &promv1a1.Route{
-			Receiver: "pagerduty",
+			Receiver: "null",
 			Routes: []apiextensionsv1.JSON{
+				convertToApiExtV1JSON(promv1a1.Route{
+					Matchers: []promv1a1.Matcher{{Name: "alertname", Value: "ClusterObjectStoreState"}},
+					Receiver: "null",
+				},
+				),
 				convertToApiExtV1JSON(promv1a1.Route{
 					GroupBy:        []string{"alertname"},
 					GroupWait:      "30s",
 					GroupInterval:  "5m",
 					RepeatInterval: "12h",
-					Matchers:       []promv1a1.Matcher{{Name: "severity", Value: "critical"}},
+					Matchers:       []promv1a1.Matcher{{Name: "alertname", Value: smtpRegxRuleMatchers, Regex: true}},
+					Receiver:       "SendGrid",
+				},
+				),
+				convertToApiExtV1JSON(promv1a1.Route{
+					GroupBy:        []string{"alertname"},
+					GroupWait:      "30s",
+					GroupInterval:  "5m",
+					RepeatInterval: "12h",
+					Matchers:       []promv1a1.Matcher{{Name: "severity", Value: pagerdutyRegxRuleMatchers, Regex: true}},
```

Comment:
This should be a name based matcher, not a severity based one as we have an exact white list of alerts for this receiver

Reply:
The Alerts that are not relevant to PagerDuty are routed to corresponding services and its not required to specify the full list of alerts that needs to routed to pagerduty. Thus picking the remaining alerts based on severity for pagerduty.